examples/osgviewerIPhone/iphoneViewerAppDelegate.mm

```diff
@@ -175,32 +199,41 @@ virtual bool handle (const osgGA::GUIEventAdapter &ea, osgGA::GUIActionAdapter &
                     ss << "Touch " << tp.id;
                     _texts[j]->setText(ss.str());
                     
+                    osg::Vec4 color;
+                    
                     switch (tp.phase)
                     {
                         case osgGA::GUIEventAdapter::TOUCH_BEGAN:
-                            _drawables[j]->setColor(osg::Vec4(0,1,0,1));
-                            std::cout << "touch began: " << ss.str() << std::endl;
+                            color = osg::Vec4(0,1,0,1);
+                            
+                            OSG_INFO << "touch began: " << ss.str() << std::endl;
                             break;
                             
                         case osgGA::GUIEventAdapter::TOUCH_MOVED:
-                            //std::cout << "touch moved: " << ss.str() << std::endl;
-                            _drawables[j]->setColor(osg::Vec4(1,1,1,1));
+                            //OSG_INFO << "touch moved: " << ss.str() << std::endl;
+                            color = osg::Vec4(1,1,1,1);
                             break;
                             
                         case osgGA::GUIEventAdapter::TOUCH_ENDED:
-                            _drawables[j]->setColor(osg::Vec4(1,0,0,1));
-                            std::cout << "touch ended: " << ss.str() << std::endl;
+                            color = osg::Vec4(1,0,0,1);
+                            OSG_INFO << "touch ended: " << ss.str() << std::endl;
                             ++num_touch_ended;
                             break;
                             
                         case osgGA::GUIEventAdapter::TOUCH_STATIONERY:
-                            _drawables[j]->setColor(osg::Vec4(0.8,0.8,0.8,1));
+                            color = osg::Vec4(0.5,0.5,0.5,1);
                             break;
                             
                         default:
                             break;
                             
                     }
+#if defined(OSG_GLES2_AVAILABLE) || defined(OSG_GLES3_AVAILABLE)
```

Comment:
This change ties this code to a specific shader implementation that uses color as a uniform.  The osg::State class has inbuilt support for aliasing fixed function vertex attributes like gl_Vertex, gl_Color etc. to osg_Vertex, osg_Color etc, my suggest would be that osg_Color should be used in the shader instead, rather than hardwiring other code to a particular shader.